base/src/com/google/idea/blaze/base/lang/buildfile/actions/BuildFileModifierImpl.java

```diff
@@ -270,9 +277,46 @@ private PsiElement createCommaElement(Project project) {
         return newElement.getPsi();
     }
 
-    private PsiElement createRule(Project project, Kind ruleKind, String ruleName) {
-    String text =
-        Joiner.on("\n").join(ruleKind.toString() + "(", "    name = \"" + ruleName + "\"", ")");
+  private static String run(String[] command) throws IOException, InterruptedException {
+    return run(command, "");
+  }
+
+  private static String run(String[] command, String input) throws IOException, InterruptedException {
+    ProcessBuilder builder = new ProcessBuilder(command);
```

Comment:
Take a look at [OSProcessHandler](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/execution/process/OSProcessHandler.java) and [OutputListener](https://github.com/JetBrains/intellij-community/blob/master/platform/lang-impl/src/com/intellij/execution/OutputListener.java) - two IntelliJ utils that give you a wrapper over process execution.

It's used in various places, I'm also using it for the Wix Bazel plugin, see: https://github.com/wix-private/wix-intellij-plugin/blob/b912dcf13faf19b72c22451c14dcfd45e80b8b25/src/main/scala/com/wixpress/execution/package.scala#L59

Reply:
Replaced ProcessBuilder with OSProcessHander